### PR TITLE
First attempt

### PR DESCRIPTION
**Pull request by aleksandr-samila**

### CODE REVIEW

File: src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs [hunk 1]
```diff
@@ -12,14 +12,15 @@ public class LoadHandlersBehaviorTests
         [Test]
         public void Should_throw_when_there_are_no_registered_message_handlers()
         {
-            var behavior = new LoadHandlersConnector(new MessageHandlerRegistry(), new InMemorySynchronizedStorage(), new InMemoryTransactionalSynchronizedStorageAdapter());
+          //TODO: Should be reviewed after InMemory move 
+            //var behavior = new LoadHandlersConnector(new MessageHandlerRegistry(), new InMemorySynchronizedStorage(), new InMemoryTransactionalSynchronizedStorageAdapter());
```
Comment (andreasohlund) [inline src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs:16]: Just create a fake InMemorySynchronizedStorage in this class and use that one

File: src/NServiceBus.Core/Persistence/InMemory/Outbox/InMemoryOutboxSettingsExtensions.cs [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
 {
     using System;
-    using Configuration.AdvancedExtensibility;
-    using Features;
     using NServiceBus.Outbox;
 
     /// <summary>
     /// Contains InMemoryOutbox-related settings extensions.
     /// </summary>
+    [ObsoleteEx(Message = "!!!", TreatAsErrorFromVersion = "8.0.0", RemoveInVersion = "9.0.0")]
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Persistence/InMemory/Outbox/InMemoryOutboxSettingsExtensions.cs:9]: Move this class to obsoletes.cs

File: src/NServiceBus.Core/Persistence/InMemory/Outbox/InMemoryOutboxPersistence.cs [hunk 3]
```diff
@@ -1,73 +1,10 @@
...
-    using System.Threading.Tasks;
-    using Microsoft.Extensions.DependencyInjection;
-    using NServiceBus.Outbox;
-
     /// <summary>
     /// Used to configure in memory outbox persistence.
     /// </summary>
-    public class InMemoryOutboxPersistence : Feature
+    [ObsoleteEx(Message = "!!!", TreatAsErrorFromVersion = "8.0.0", RemoveInVersion = "9.0.0")]
+    public class InMemoryOutboxPersistence
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Persistence/InMemory/Outbox/InMemoryOutboxPersistence.cs:7]: Move this class to obsoletes.cs

File: src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersistence.cs [hunk 4]
```diff
@@ -3,20 +3,8 @@
     /// <summary>
     /// Used to configure in memory saga persistence.
     /// </summary>
-    public class InMemorySagaPersistence : Feature
+    [ObsoleteEx(Message = "!!!", TreatAsErrorFromVersion = "8.0.0", RemoveInVersion = "9.0.0")]
+    public class InMemorySagaPersistence
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersistence.cs:7]: Move this class to obsoletes.cs

File: src/NServiceBus.Core/Persistence/InMemory/SubscriptionStorage/InMemorySubscriptionPersistence.cs [hunk 5]
```diff
@@ -3,19 +3,8 @@
     /// <summary>
     /// Used to configure in memory subscription persistence.
     /// </summary>
-    public class InMemorySubscriptionPersistence : Feature
+    [ObsoleteEx(Message = "!!!", TreatAsErrorFromVersion = "8.0.0", RemoveInVersion = "9.0.0")]
+    public class InMemorySubscriptionPersistence
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Persistence/InMemory/SubscriptionStorage/InMemorySubscriptionPersistence.cs:7]: Move this class to obsoletes.cs

File: src/NServiceBus.Core/Persistence/InMemory/TimeoutPersister/InMemoryTimeoutPersistence.cs [hunk 6]
```diff
@@ -1,23 +1,10 @@
 ﻿namespace NServiceBus.Features
 {
-    using System;
-
     /// <summary>
     /// Used to configure in memory timeout persistence.
     /// </summary>
-    public class InMemoryTimeoutPersistence : Feature
+    [ObsoleteEx(Message = "!!!", TreatAsErrorFromVersion = "8.0.0", RemoveInVersion = "9.0.0")]
+    public class InMemoryTimeoutPersistence
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Persistence/InMemory/TimeoutPersister/InMemoryTimeoutPersistence.cs:7]: Move this class to obsoletes.cs

File: src/NServiceBus.Core/Persistence/InMemory/InMemory.cs [hunk 7]
```diff
@@ -1,19 +1,10 @@
...
 {
-    using Features;
-    using Persistence;
-
     /// <summary>
     /// Used to enable InMemory persistence.
     /// </summary>
-    public class InMemoryPersistence : PersistenceDefinition
+    [ObsoleteEx(Message = "!!!", TreatAsErrorFromVersion = "8.0.0", RemoveInVersion = "9.0.0")]
+    public class InMemoryPersistence
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Persistence/InMemory/InMemory.cs:7]: Move this class to obsoletes.cs

**Review by andreasohlund — CHANGES_REQUESTED** [review]

**Comment by adamralph** [comment]
@lailabougria can we delete the branch?

**Comment by lailabougria** [comment]
@adamralph Done!

**Pull request closed without merging**